src/coreclr/jit/emit.cpp

```diff
@@ -4479,6 +4610,429 @@ void emitter::emitJumpDistBind()
+//     the igOffs, igSize and emitTotalCodeSize.
+//
+void emitter::emitLoopAlignAdjustments()
+{
+    // no align instructions
+    if (emitAlignList == nullptr)
+    {
+        return;
+    }
+
+    JITDUMP("*************** In emitLoopAlignAdjustments()\n");
+
+    unsigned short estimatedPaddingNeeded, alignmentBoundary = emitComp->opts.compJitAlignLoopBoundary;
+    unsigned       maxLoopSize = 0;
+
+    if (emitComp->opts.compJitAlignLoopAdaptive)
+    {
+        // For adaptive, adjust the loop size depending on the alignment boundary
+        int maxBlocksAllowedForLoop = genLog2((unsigned)alignmentBoundary) - 1;
+        maxLoopSize                 = alignmentBoundary * maxBlocksAllowedForLoop;
+        estimatedPaddingNeeded      = (alignmentBoundary >> 1) - 1;
+    }
+    else
+    {
+        // For non-adaptive, just take whatever is supplied using COMPlus_ variables
+        maxLoopSize            = emitComp->opts.compJitAlignLoopMaxCodeSize;
+        estimatedPaddingNeeded = alignmentBoundary - 1;
+    }
+
+    unsigned        alignBytesRemoved = 0;
+    unsigned        loopSize          = 0;
+    unsigned        loopIGOffset      = 0;
+    instrDescAlign* alignInstr        = emitAlignList;
+
+    // track the IG that was adjusted so we can update the offsets
+    insGroup* lastIGAdj = emitAlignList->idaIG;
+
+    for (; alignInstr != nullptr; alignInstr = alignInstr->idaNext)
+    {
+        assert(alignInstr->idIns() == INS_align);
+
+        insGroup* alignIG = alignInstr->idaIG;
+
+        // Adjust offsets of all IGs until the current IG
+        while (lastIGAdj->igNum <= alignIG->igNum)
+        {
+            lastIGAdj->igOffs -= alignBytesRemoved;
+            lastIGAdj = lastIGAdj->igNext;
+        }
+
+        loopIGOffset = alignIG->igOffs + alignIG->igSize;
+
+        // igSize also includes INS_align instruction, take it off.
+        loopIGOffset -= estimatedPaddingNeeded;
+
+        // IG can be marked as not needing alignment if during setting igLoopBackEdge, it is detected
+        // that the igLoopBackEdge covers an IG that is marked for alignment.
+        unsigned actualPaddingNeeded =
+            alignIG->isLoopAlign() ? emitCalculatePaddingForLoopAlignment(alignIG, loopIGOffset DEBUG_ARG(false)) : 0;
+
+        assert(estimatedPaddingNeeded >= actualPaddingNeeded);
+
+        unsigned short diff = (unsigned short)(estimatedPaddingNeeded - actualPaddingNeeded);
+
+        if (diff != 0)
+        {
+            alignIG->igSize -= diff;
+            alignBytesRemoved += diff;
+            emitTotalCodeSize -= diff;
+
+            // Update the flags
+            alignIG->igFlags |= IGF_UPD_ISZ;
+            if (actualPaddingNeeded == 0)
+            {
+                alignIG->igFlags &= ~IGF_LOOP_ALIGN;
+            }
+
+            if (emitComp->opts.compJitAlignLoopAdaptive)
+            {
+                assert(actualPaddingNeeded < 15);
```

Comment:
Update the `15` to use your new symbolic constant.

Reply:
good point.